saga/sagaState.go

```diff
@@ -6,30 +6,40 @@ import (
 )
 
 /*
+ * Additional data about tasks to be committed to the log
+ * This is Opaque to SagaState, but useful data to persist for
+ * results or debugging
+ */
+type taskData struct {
```

Comment:
I'm not concerned about an extra string copy per task; I just thought it might be nice to put all the task state and data together, so it's:
map of taskId -> taskState

Then per-task methods could just take a taskState, instead of both a flag and taskData. Would that be useful or not?

The immutable comment just meant doing what you're doing, not any kind of hardening (which, yes, Go doesn't provide).